src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,297 @@
 
 /**
- * Google Analytics for Firebase is a free app measurement solution that provides insight on
- * app usage and user engagement.
+ * Google Analytics for Firebase is a free app measurement solution that provides insight on app
+ * usage and user engagement.
  *
  * @see <a href="https://firebase.google.com/docs/analytics/">Google Analytics for Firebase</a>
  * @see <a href="#">Google Analytics for Firebase Integration</a>
  * @see <a href="#">Google Analytics for Firebase Android SDK</a>
  */
 public class FirebaseIntegration extends Integration<FirebaseAnalytics> {
 
-  public static FirebaseAnalytics mFirebaseAnalytics;
+  public static final Factory FACTORY =
+      new Factory() {
+        @Override
+        public Integration<?> create(ValueMap settings, Analytics analytics) {
+          Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
+          if (!hasPermission(
+              analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
+            logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
+            return null;
+          }
+          if (!hasPermission(analytics.getApplication(), Manifest.permission.INTERNET)) {
+            logger.debug("INTERNET is required for Firebase Analytics.");
+            return null;
+          }
+          if (!hasPermission(analytics.getApplication(), Manifest.permission.WAKE_LOCK)) {
+            logger.debug("WAKE_LOCK is required for Firebase Analytics.");
+            return null;
+          }
 
-  public static final Factory FACTORY = new Factory() {
-    @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-      Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
-      if (!hasPermission(analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
-        logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
-        return null;
-      }
-      if (!hasPermission(analytics.getApplication(), Manifest.permission.INTERNET)) {
-        logger.debug("INTERNET is required for Firebase Analytics.");
-        return null;
-      }
+          Context context = analytics.getApplication();
 
-      Context context = analytics.getApplication();
-
-      return new FirebaseIntegration(context);
-    }
+          return new FirebaseIntegration(context, logger);
+        }
 
-    @Override public String key() {
-      return FIREBASE_ANALYTICS_KEY;
-    }
-  };
+        @Override
+        public String key() {
+          return FIREBASE_ANALYTICS_KEY;
+        }
+      };
 
-  private static final String FIREBASE_ANALYTICS_KEY = "Firebase Analytics";
+  static final Pattern ISO_DATE =
+          Pattern.compile("(^([0-9]{4})-(1[0-2]|0[1-9])-(0[1-9]|1[1-9]|2[0-9]|3[0-1]).*)", CASE_INSENSITIVE);
+  private static final String FIREBASE_ANALYTICS_KEY = "Firebase";
+  final Logger logger;
+  final FirebaseAnalytics mFirebaseAnalytics;
```

Comment:
This should just be `firebaseAnalytics `. We don't use the hugarian notation. Also it's inconsistent with the other member variables.